boot/seal.go

```diff
@@ -654,7 +654,11 @@ func recoveryBootChainsForSystems(systems []string, modesForSystems map[string][
 		for _, system := range systems {
 			// get kernel and gadget information from seed
 			perf := timings.New(nil)
-			seedSystemModel, snaps, err := seedReadSystemEssential(dirs.SnapSeedDir, system, []snap.Type{snap.TypeKernel, snap.TypeGadget}, perf)
+			seedDir := dirs.SnapSeedDir
+			if osutil.FileExists(filepath.Join(dirs.GlobalRootDir, "etc/initrd-release")) {
```

Comment:
I'm not entirely a fan of this as we have other places that have the same issue and solve it differently and this is very implicit. Some kind of flag or path coming down through boot.MakeRunnableStandaloneSystem perhaps?

Reply:
I have added one `MakeRunnable...` function that propagates the path down to there.